guesser.rb

```diff
@@ -0,0 +1,45 @@
+def guesser
+    loop do # Outer loop to restart game
+        puts "Think of a number between 1 and 100"
+        puts "I will guess. If I am too high, type H; if I am too low, type L."
+        puts "If I am correct, type Y."
+
+        low_num = 1
+        high_num = 100
+        guesses = []    # Array to hold guesses
+        
+        loop do # Inner loop with guessing logic
+            guess = (high_num + low_num)/2  # Chooses a number in the middle of the current range
+
+            if guesses.include?(guess)  # Checks if a guess is repeated to determine if player is lying
+                puts "You have deceived me; please be truthful next time."
+                puts "These are my guesses :  #{guesses}. I know one of them is correct."
+                break
+            end
+            
+            guesses << guess    # Adds guess to guesses array
+
+            puts "Is the number #{guess}?"
+            response = gets.chomp.upcase
+
+            # Gets responses from player to determine next guess
+            if response == 'H'
+                high_num = guess - 1
+            elsif response == 'L'
+                low_num = guess + 1
+            elsif response == 'Y'
+                puts "I knew I had it."
+                break
+            else
+                "Please enter the appropriate response."
```

Comment:
❗ You are missing the `puts` here.
❗ Also, because you are storing `guesses`, when this `else` case cause the loop to restart, the game accuses the player of cheating because `guesses` already includes the `guess`:

```ruby
irb(main):847:0> guesser # me, invoking your method
Think of a number between 1 and 100
I will guess. If I am too high, type H; if I am too low, type L.
If I am correct, type Y.
Is the number 50?
y
I knew I had it.
Do you want to play again? (yes/no)
yes
Think of a number between 1 and 100
I will guess. If I am too high, type H; if I am too low, type L.
If I am correct, type Y.
Is the number 50?
n # an invalid response, no warning because puts is missing
You have deceived me; please be truthful next time.
These are my guesses :  [50]. I know one of them is correct.
```

This is a good reason to consider changing your cheating logic to merely consider if `high_num` is lower than `low_num` as I suggested above.